### ISSUE TITLE
[popups] Visually hidden close buttons

### ISSUE THREAD

**Issue by atomiks** [mui/base-ui#2722]
# Feature request

Right now, the `modal` prop doesn't trap focus except for `Dialog` — it's allowed to escape `Menu`, `Select`, `Popover`, etc as they often don't use visual close buttons. This is done so screen readers don't remain trapped (mainly touch, where `Esc` is not available to dismiss). 

Furthermore, if `Close` isn't rendered for Dialog, the same issue occurs.

Focus could be trapped in modal popups if the `Close` part is detected inside the popup. 

Although `aria-label` gets translated in browsers, this can't necessarily be done automatically if the webpage isn't expected to be translated because it has built-in i18n: https://www.openaccess.nz/blog/do-aria-attributes-get-translated/

```jsx
<Popover.Close aria-label="Dismiss popup" visuallyHidden />
```

`Combobox` could also benefit from this particularly for the **Multiple select** demo, where a touch screen reader is trapped when swiping backwards until focus lands on a chip  (which closes the menu, removing `aria-hidden` from outside elements). It is missing a `Close` part entirely.
